### ISSUE TITLE
Use Mazsi's new description of the circular buffer

### ISSUE THREAD

**Issue by joefowler** [usnistgov/dastard#167]
Mazsi wants to update the buffer descriptor for the ring buffer to read as follows (in C):

```c
/// Buffer description information: lives in shared memory
typedef struct {
   uint32_t magic;
   uint32_t version;
   uint64_t write_index;
   uint64_t read_index;
   uint64_t buffer_size;
   uint64_t packet_size;
} __attribute__((packed,aligned(64))) buffer_description_t;
```

Notes:
* Exact magic number is TBD. We'd just need to ensure that it changes under byte-order swapping.
* Use semantic versioning for the `version` item.
* `packet_size` will be a constant. 0 will signify something non-packetized, perhaps the stream we have now.

**Comment by joefowler** [comment]
See [Mazsi's writer](https://gitlab.nist.gov/gitlab/mazsi/circbtest/blob/master/buffer-shmem.c#L23). It uses **0xb0ffde5c** as the magic number

**Comment by joefowler** [comment]
Fixed in 3d933cd7 (see also PR #168).
